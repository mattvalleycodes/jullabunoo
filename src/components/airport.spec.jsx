import * as React from "react";
import sinon from "sinon";
import renderer from "react-test-renderer";

import Airport from "./airport";

describe("<Airport /> component", () => {
  const sandbox = sinon.createSandbox();

  beforeEach(() => {
    // Thu Jan 23 2020 22:13:52 GMT+1100 (Australian Eastern Daylight Time)
    const now = 1579778039119;
    sandbox.useFakeTimers(now);
  });

  afterEach(() => {
    sandbox.restore();
  });

  test("renders the airport information", () => {
    const airport = renderer
      .create(
        <Airport
          code="BEK"
          name="Rae Bareli"
          city="Rae Bareli City"
          country="India"
          region="Asia"
          timezone="Australia/Sydney"
        />,
      )
      .toJSON();

    expect(airport).toMatchSnapshot();
  });
});
