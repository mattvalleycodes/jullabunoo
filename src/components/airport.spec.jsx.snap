// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Airport /> component renders the airport information 1`] = `
<div
  className="box airport"
>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      Code
      :
    </b>
    <span>
      BEK
    </span>
  </div>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      Name
      :
    </b>
    <span>
      Rae Bareli
    </span>
  </div>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      City
      :
    </b>
    <span>
      Rae Bareli City
    </span>
  </div>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      Country
      :
    </b>
    <span>
      India
    </span>
  </div>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      Region
      :
    </b>
    <span>
      Asia
    </span>
  </div>
  <div
    className="data"
  >
    <b
      className="data__title"
    >
      Local time
      :
    </b>
    <span>
      Thursday, January 23rd 2020, 10:13:59 pm
    </span>
  </div>
  <br />
  <a
    aria-current="page"
    href="/"
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;
